runtimes/pythonrt/worker_grpc_handler.go

```diff
@@ -373,6 +375,37 @@ func (s *workerGRPCHandler) Unsubscribe(ctx context.Context, req *pb.Unsubscribe
 	}
 }
 
+func (s *workerGRPCHandler) EncodeJWT(ctx context.Context, req *pb.EncodeJWTRequest) (*pb.EncodeJWTResponse, error) {
+	// GitHub's JWTs must be signed using the RS256 algorithm:
+	// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+	if req.Algorithm != jwt.SigningMethodRS256.Name {
+		return &pb.EncodeJWTResponse{Error: fmt.Sprintf("unsupported signing method: %s", req.Algorithm)}, nil
+	}
+
+	claims := jwt.MapClaims{}
+	for key, value := range req.Payload {
+		claims[key] = value
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+
+	pem, ok := os.LookupEnv("GITHUB_PRIVATE_KEY")
```

Comment:
are you sure this is available at the runner?

Reply:
As long as the runner is a part of the AK server - yes. This is a server-wide setting, not connection/runner-specific.

Reply:
fyi @efiShtain